src/UI/Validation.java

```diff
@@ -12,7 +12,7 @@
 	 * @return true if the card number is all digits and 13-16 in length
 	 */
 	public static boolean validateCardNumber(String cardNumber) {
-		return Pattern.matches("^\\d{13,16}$", cardNumber);
+		return Pattern.matches("^\\d{13,16}$", cardNumber) || cardNumber.trim().equals("");
```

Comment:
wait, why do we want it to equal the empty string? Isn't that bad?? same comment for other methods

Reply:
My bad, I forgot to mention this. When I was trying to modify cards and didn't want to change everything, the ones that were left blank kept getting caught by the validation method so I just made the validation method accept empty strings since we already had other statements on the front end that checked for empty entries in the places that weren't allowed

Reply:
ok cool can you just write that in a comment in each method so that when grading they understand? feel like this is a good usage for a good comment